index.js

```diff
@@ -95,20 +147,25 @@ function testGenResponse(swagger, path, operation, response, config) {
  * @param  {string} path API path to generate tests for
  * @param  {string} operation operation of the path to generate tests for
  * @param  {json} config configuration for testGen
+ * @param  {json} parameters json for different parameters
  * @returns {string|Array} set of all tests for a path's operation
  */
-function testGenOperation(swagger, path, operation, config) {
+function testGenOperation(swagger, path, operation, config, parameters) {
   var responses = swagger.paths[path][operation].responses;
   var result = [];
   var res;
+  var origin = JSON.parse(JSON.stringify(parameters));
 
   for (res in responses) {
     if (responses.hasOwnProperty(res)) {
-      result.push(testGenResponse(swagger,
-        path,
-        operation,
-        res,
-        config));
+      parameters = JSON.parse(JSON.stringify(origin));
```

Comment:
The reason I did that is because, I want to get the copy of parameters, not the reference. So I don't want to parameters get repeated after each loop, so I have to get the original parameters, if I only do origin = parameters, it will be the reference. so I did: 
var origin = JSON.parse(JSON.stringify(parameters));
 parameters = JSON.parse(JSON.stringify(origin));

Mohsen, do you have any better tips ?